src/componentProvider.js

```diff
@@ -0,0 +1,70 @@
+import componentLoader from './componentLoader';
+
+/**
+ * Compile component elements and DOM attachments.
+ *
+ * @param {Object} config The component config.
+ * @returns {Function} A function to initialize the component.
```

Comment:
From the ECMAScript2015 spec

> **4.3.10 undefined value**
> 
> primitive value used when a variable has not been assigned a value
> 
> **4.3.12 null value**
> 
> primitive value that represents the intentional absence of any object value

A function with no specified return value implicitly returns null|void because there is no assignment of a value, there is an intentional absence of a value. Functions are just objects.

You could just `return;`, so no need to `return undefined;`

Reply:
> `undefined` is the default return value for functions

Put another way, a variable assigned to a function that has no return value will be `undefined`. The only circumstance for which `componentProvider` should return a value is if `config.load === false`. Adding to that, [ESLint wants a consistent return](https://eslint.org/docs/latest/rules/consistent-return), and their docs do a better job of conveying my initial response.